engineering_tools/carma_script_extensions/cloud-sim

```diff
@@ -0,0 +1,348 @@
+        --capabilities CAPABILITY_NAMED_IAM)
+    
+    local VALID_LAUNCH=$(echo "${RESULT}" | grep 'arn:aws:cloudformation:us-east-1')
+    if [ -z "${VALID_LAUNCH}" ]; then
+        echo "AWS Launch Failed with error:"
+        echo ${RESULT}
+        exit 1
+    fi
+    
+    echo "AWS Stack ${STACK_NAME} starting up......."
+
+    local STACK_RESOURCE_STATUS="NOT_EMPTY"
+    local LAUNCHED_RESOURCE_COUNT=0
+    local PREV_LAUNCH_COUNT=0
+    local i=0
+    local SLEEP_SEC=5
+    local LAUNCH_TIMEOUT=595 # 10m timeout to launch the stack
+    local ELAPSED_TIME=0
+    local NON_COMPLETE_RES="ALL"
+    local FAILED_RES="NONE"
+    local TOTAL_STACK_RESOURCE=0
+
+    sleep ${SLEEP_SEC}
+
+    # As long as there are resources which do not have the status CREATE_COMPLETE 
+    while [ -n "${NON_COMPLETE_RES}" ] || [ ${ELAPSED_TIME} -gt ${LAUNCH_TIMEOUT} ] ; do 
+
+        STACK_RESOURCE_STATUS=$(aws cloudformation describe-stack-resources --stack-name "${STACK_NAME}" | grep -oh '"ResourceStatus":.*",' | sed "s|\s*\"ResourceStatus\": \"||" | sed "s|\",||")
+
+        NON_COMPLETE_RES=$(echo "${STACK_RESOURCE_STATUS}" | grep -v "CREATE_COMPLETE" | sed "s/\n//" )
+        FAILED_RES=$(echo "${STACK_RESOURCE_STATUS}" | grep "\(ROLLBACK\|FAILED\|DELETE\)" | sed "s/\n//")
+
+        # If the status returns a ROLLBACK or FAILED state at any point then launch has failed
+        if [ -n "${FAILED_RES}" ]; then 
+            echo "Stack launch failed. Inspect AWS console for details. Remember to delete the stack after debugging!"
+            exit 1
+        fi
+
+        TOTAL_STACK_RESOURCE=$(echo "${STACK_RESOURCE_STATUS}" | wc -l)
+        
+        LAUNCHED_RESOURCE_COUNT=$(echo "${STACK_RESOURCE_STATUS}" | grep "CREATE_COMPLETE" | wc -l)
+
+        if [ ${PREV_LAUNCH_COUNT} -ne ${LAUNCHED_RESOURCE_COUNT} ]; then
+            echo "Launched Stack Resource: ${LAUNCHED_RESOURCE_COUNT}/${TOTAL_STACK_RESOURCE} Elapsed Sec: ${ELAPSED_TIME}"
+            
+            PREV_LAUNCH_COUNT=${LAUNCHED_RESOURCE_COUNT}
+
+        elif [ $(( ELAPSED_TIME % 20 )) == 0 ]; then
+            echo "Waiting on resources. Launched Stack Resources: ${LAUNCHED_RESOURCE_COUNT}/${TOTAL_STACK_RESOURCE} Elapsed Sec: ${ELAPSED_TIME}"
+        fi 
+
+        ((i++))
+
+        # Wait SLEEP_SEC seconds before checking again
+        sleep ${SLEEP_SEC}
+        ELAPSED_TIME=$((SLEEP_SEC * i))
+
+    done
+
+    if [ ${ELAPSED_TIME} -gt ${LAUNCH_TIMEOUT} ]; then
+        echo "Stack launch timedout. Inspect AWS console for details. Remember to delete the stack after debugging!"
+        exit 1
+    fi
+
+
+    local DEV_MACHINE_ID=$(aws cloudformation describe-stack-resources --stack-name ${STACK_NAME} --output text --logical-resource-id DevMachine --query StackResources[].PhysicalResourceId)
+
+    local PUBLIC_DNS=$(aws ec2 describe-instances --instance-ids ${DEV_MACHINE_ID} --output text --query Reservations[].Instances[].PublicDnsName)
+
+    # TODO determine cloud 9 address
```

Comment:
Still relevant TODO or can we remove this?

Reply:
This was completed, I'll remove